### PR TITLE
copr/metrics: add request wait time

### PR DESCRIPTION
**Pull request by BusyJay**
Requests may wait too long in queue before actually being handled, this pr add a metric to trace this problem.

@siddontang @disksing @shenli PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
@siddontang @zhangjinpeng1987 PTAL

File: src/server/coprocessor/endpoint.rs [hunk 1]
```diff
@@ -105,20 +105,60 @@ impl Display for Task {
...
+            return;
+        }
+        let wait_time = duration_to_sec(self.timer.elapsed());
+        COPR_REQ_WAIT_TIME.with_label_values(&["select", get_req_type_str(self.req.get_tp())])
+            .observe(wait_time);
+        self.wait_time = Some(wait_time);
+    }
+
+    fn finish_handling(&mut self) {
+        self.start_handling();
```
Comment (zhangjinpeng1987) [inline src/server/coprocessor/endpoint.rs]: why call `start_handling` here?

Comment (BusyJay) [inline src/server/coprocessor/endpoint.rs]: In case `start_handling` is not called yet. Calling it here to update wait time.

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/server/coprocessor/endpoint.rs (hunk 1 above)
Comment (siddontang) [inline src/server/coprocessor/endpoint.rs]: I think we should think a better name. Start is confused here

**Comment by BusyJay** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

PTAL @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**